[
    {
        "content": "<p>Dear community,</p>\n<p>I am currently setting up a hapi-server for a feasibility study. <br>\nTherefore we want to limit access by setting a simple user login.<br>\nIs it possible to set a user name and a password when running a hapi server with docker?</p>\n<p>Thanks in advance for your answers.</p>",
        "id": 274669993,
        "sender_full_name": "Mareike Przysucha",
        "timestamp": 1646822780
    },
    {
        "content": "<p>Not included in the docker image. You could:</p>\n<ul>\n<li>modify JPAstarter and add a BasicAuth Interceptor</li>\n<li>put an apache/nginx reverse proxy in front of hapi and do auth in the apache (no need to modifiy the docker image)</li>\n</ul>",
        "id": 274672624,
        "sender_full_name": "Patrick Werner",
        "timestamp": 1646824209
    },
    {
        "content": "<p>Thanks, Patrick, for your reply. I will try and report on my experiances.</p>",
        "id": 274673433,
        "sender_full_name": "Mareike Przysucha",
        "timestamp": 1646824684
    },
    {
        "content": "<p>Hi there. Due to our requirements I think only the first is possible.<br>\nNow I am facing the next problem:<br>\nI wrote a BasicSecurityInterceptor like proposed <a href=\"https://hapifhir.io/hapi-fhir/docs/security/introduction.html\">here</a> - though changing the verification part - and added the interceptor in BaseJpaRestfulServer with <code>registerInterceptor(new BasicSecurityInterceptor());</code>. When I then build the docker image and run this image, I get the following BeanDefinitionOverrideException:</p>\n<div class=\"codehilite\"><pre><span></span><code>Caused by: org.springframework.beans.factory.support.BeanDefinitionOverrideException:\nInvalid bean definition with name &#39;mdmRuleValidator&#39; defined in class path resource\n[ca/uhn/fhir/jpa/starter/mdm/MdmConfig.class]:\nCannot register bean definition [Root bean: class [null]; scope=; abstract=false; lazyInit=null;\nautowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false;\nfactoryBeanName=mdmConfig; factoryMethodName=mdmRuleValidator; initMethodName=null;\ndestroyMethodName=(inferred);\ndefined in class path resource [ca/uhn/fhir/jpa/starter/mdm/MdmConfig.class]]\nfor bean &#39;mdmRuleValidator&#39;:\nThere is already [Root bean: class [null]; scope=; abstract=false; lazyInit=null; autowireMode=3;\ndependencyCheck=0; autowireCandidate=true; primary=false;\nfactoryBeanName=ca.uhn.fhir.jpa.mdm.config.MdmCommonConfig;\nfactoryMethodName=mdmRuleValidator; initMethodName=null; destroyMethodName=(inferred);\ndefined in class path resource [ca/uhn/fhir/jpa/mdm/config/MdmCommonConfig.class]]\nbound.\n</code></pre></div>\n<p>Does anyone have an idea on how to cope with this problem?</p>",
        "id": 275372736,
        "sender_full_name": "Mareike Przysucha",
        "timestamp": 1647351042
    },
    {
        "content": "<p>your error points to the mdm part of hapi. Did you change sth. for the MDM Config?</p>",
        "id": 275379061,
        "sender_full_name": "Patrick Werner",
        "timestamp": 1647354077
    },
    {
        "content": "<p>not that I were aware of, but I'll check it.</p>",
        "id": 275381025,
        "sender_full_name": "Mareike Przysucha",
        "timestamp": 1647354966
    },
    {
        "content": "<p>either enable <a href=\"https://github.com/trifork/hapi-fhir-jpaserver-starter/blob/929a3535fa93d33b24cef06ed4158ae2fcb9f6a2/src/main/resources/application.yaml#L4\">https://github.com/trifork/hapi-fhir-jpaserver-starter/blob/929a3535fa93d33b24cef06ed4158ae2fcb9f6a2/src/main/resources/application.yaml#L4</a> or disable <a href=\"https://github.com/trifork/hapi-fhir-jpaserver-starter/blob/929a3535fa93d33b24cef06ed4158ae2fcb9f6a2/src/main/resources/application.yaml#L103\">https://github.com/trifork/hapi-fhir-jpaserver-starter/blob/929a3535fa93d33b24cef06ed4158ae2fcb9f6a2/src/main/resources/application.yaml#L103</a></p>",
        "id": 275381219,
        "sender_full_name": "Jens Villadsen",
        "timestamp": 1647355071
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"191451\">@Patrick Werner</span> : I copied the files from jpa-starter =&gt; same error. That's not the problem.<br>\n<span class=\"user-mention\" data-user-id=\"191427\">@Jens Villadsen</span>: I enabled the first, now there is no more error. Thank you so much.</p>",
        "id": 275382400,
        "sender_full_name": "Mareike Przysucha",
        "timestamp": 1647355527
    }
]